Dockerfile

```diff
@@ -38,5 +38,7 @@ RUN npm run build
 # Production
 FROM base AS production
 
+CMD ["npm", "run", "start"]
```

Comment:
For a production image it's best to call 'node index.js' directly rather than use npm so the application can response to docker signals.

Reply:
Interesting. I think it's actually best not to call node directly but rather use a "dumb init" script (still not what I'm doing here) but I'll look into it next week.

Reply:
I've done some testing around this and found that while most signals (e.g. SIGTERM and SIGINT) are passed correctly through npm to the node process, SIGHUP seems to not make it through so it's definitely better not to run through `npm start` as you say, @markysoft.

I've also refreshed my memory on dumb init scripts, which are recommended to ensure that containerised processes do not run as PID 1 (which has special rules in UNIX systems, so doesn't always react as expected to process signals). Docker itself actually has a built-in init script these days, which can be enabled via the `docker run --init` flag or an `init` property in `docker-compose.yml`. For Kubernetes, I haven't seen any mention of a built-in solution so I think it's best we build something into our container images.

I've implemented this using [Tini](https://github.com/krallin/tini), which is the basis of the built-in Docker solution and works transparently with existing images - you just install via `apk` and set `tini --` as the `ENTRYPOINT` in the `Dockerfile`.